decode.go

```diff
@@ -355,6 +356,14 @@ func (d *Decoder) decodeValue(dst reflect.Value, src ast.Node) error {
 			return errors.Wrapf(err, "failed to UnmarshalYAML")
 		}
 		return nil
+	} else if _, ok := dst.Addr().Interface().(*time.Time); ok {
+		return d.decodeTime(dst, src)
```

Comment:
:memo: time.Time implements encofing.TextUnmarshaler
So this should be processed before process encofing.TextUnmarshaler